gdpr/impl_test.go

```diff
@@ -177,6 +177,73 @@ func TestMalformedConsent(t *testing.T) {
 }
 
 func TestAllowPersonalInfo(t *testing.T) {
+	tests := []struct {
+		description string
+		bidderName  openrtb_ext.BidderName
+		publisherID string
+		gdpr        GDPRState
+		consent     string
+		allowPI     bool
+	}{
+		{
+			description: "Allow PI - Non standard publisher",
+			bidderName:  openrtb_ext.BidderAppnexus,
+			publisherID: "appNexusAppID",
+			gdpr:        YesGDPR,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     true,
+		},
+		{
+			description: "Don't allow PI - No GDPR",
+			bidderName:  openrtb_ext.BidderPubmatic,
+			gdpr:        NoGDPR,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     false,
+		},
+		{
+			description: "Don't allow PI - known vendor without TCF1 purposes",
+			bidderName:  openrtb_ext.BidderAppnexus,
+			gdpr:        YesGDPR,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     false,
+		},
+		{
+			description: "Allow PI - Known vendor with TCF1 purposes",
+			bidderName:  openrtb_ext.BidderPubmatic,
+			gdpr:        YesGDPR,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     true,
+		},
+		{
+			description: "Don't allow PI - default vendor with non-empty consent",
+			bidderName:  openrtb_ext.BidderRubicon,
+			gdpr:        YesGDPR,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     false,
+		},
+		{
+			description: "PI allowed according to host settings - default vendor with empty consent",
```

Comment:
What is the host setting that allows PI when GDPR applies and no consent is given?

Reply:
`UserSyncIfAmbiguous` is the setting I was referring to.

Reply:
`UserSyncIfAmbiguous` should only come into play if `AmbiguousGDPR`. If the GDPR setting is not ambiguous, then we should not be using the "IfAmbiguous" setting for anything.

Reply:
Makes sense. I didn't change the default vendor permissions logic but it seems I need to.

If the bidder is not in the vendor list, we fall back to the default vendor permissions. Should the permissions be modified to include the `&& gdpr == AmbiguousGDPR` condition as shown here?
```if consent == "" && gdpr == AmbiguousGDPR {
	return p.cfg.UsersyncIfAmbiguous, p.cfg.UsersyncIfAmbiguous, p.cfg.UsersyncIfAmbiguous, nil
}
return false, false, false, nil```

Reply:
that seems correct